### ISSUE TITLE
strange culling issue

### ISSUE THREAD

**Issue by ianmckenna87** [godotengine/godot#100155]
### Tested versions

4.4 dev6

### System information

Godot v4.4.dev6 - Windows 10.0.22631 - Multi-window, 1 monitor - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 4060 (NVIDIA; 32.0.15.6614) - 12th Gen Intel(R) Core(TM) i5-12400F (12 threads)

### Issue description

I've been racking my brains regarding a strange face culling issue I'm having. I'm using a model that has asymmetrical parts and the meshes on one side display a culling issue, with the front faces being culled, despite the normals being set (and recalculated) correctly in Blender

the meshes that appear on the right appear correctly culled
![Image](https://github.com/user-attachments/assets/3f9c0c97-fe2b-43ed-b3fc-bd681a764c74)

but on the left side they're not
![Image](https://github.com/user-attachments/assets/74f57dc3-2fe3-482f-a47d-51844cea0584)

The modeler doesn't know what could be causing it but assumes it's a shader bug in Godot.

### Steps to reproduce

project attached with model displaying the behaviour. Normals have been checked in Blender

### Minimal reproduction project (MRP)

[culling_issue_project.zip](https://github.com/user-attachments/files/18049801/culling_issue_project.zip)

**Comment by clayjohn** [comment]
I took a look and the material seems fine. The left leg appears to be using an negative scale in the animation which will flip the model inside out as you can see. Although I could be wrong about that, I am not that familiar with animation code.

When a mesh instance uses a negative scale, we can detect it and flip the model right side out so it appears to just work. But if this is inverting the mesh in the animation, we aren't able to detect and flip things automatically.

**Comment by ianmckenna87** [comment]
> I took a look and the material seems fine. The left leg appears to be using an negative scale in the animation which will flip the model inside out as you can see. Although I could be wrong about that, I am not that familiar with animation code.
> 
> When a mesh instance uses a negative scale, we can detect it and flip the model right side out so it appears to just work. But if this is inverting the mesh in the animation, we aren't able to detect and flip things automatically.

that would describe the issue I'm having more accurately as the mesh appearing inside out, but I couldn't see any scale tracks on the animation track

**Comment by clayjohn** [comment]
I looked directly at the bone matrices and they have negative scales

**Comment by ianmckenna87** [comment]
> I looked directly at the bone matrices and they have negative scales

I took a look at the skeleton itself and you are indeed correct, but not the ones that are affected. oddly enough, setting those bone scales to negative makes them appear properly, albeit upside down.

**Comment by ydeltastar** [comment]
The normals are inverted in Blender too when importing the glTF.

![Image](https://github.com/user-attachments/assets/11887286-f9d9-4aaa-a533-91f96567a3e0)

**Comment by ianmckenna87** [comment]
> The normals are inverted in Blender too when importing the glTF.
> 
> ![Image](https://github.com/user-attachments/assets/11887286-f9d9-4aaa-a533-91f96567a3e0)

Yes, you are right, I found the same thing, but, how strange, the source file normals appear fine:

![Image](https://github.com/user-attachments/assets/78ff640c-7097-458b-86bc-e43da27318a1)
